### PR TITLE
Changes required to support ppc64le

### PR DESCRIPTION
**Pull request by vrushaliinamdar**
Made following changes to make it work on ppc64le systems:
1. Created a ppc64le docker files for "pfe" and "performance" components
2. Changed start.sh to download ppc64le specific binary of cwctl when ARCH is ppc64le
3. Created a build-appsody-controller.sh to download ppc64le specific appsody binary and build appsody-controller from source.
4. Changed build.sh to use build-appsody-controller.sh in case of ppc64le architecture.

### CODE REVIEW

**Comment by cccanderson** [comment]
Please follow the first two steps on https://wiki.eclipse.org/Codewind_GitHub_Workflows in order for us to get the eclipsefdn/eca test to pass.

File: src/pfe/Dockerfile_ppc64le [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+RUN yum install -y epel-release \
+    && yum -y --enablerepo=epel install zip unzip sudo ca-certificates openssl  wget gcc-c++ make slirp4netns.rpm 
+
+
+RUN wget https://nodejs.org/download/release/latest-v13.x/node-v13.5.0-linux-ppc64le.tar.gz \
+    && tar xvzf  node-v13.5.0-linux-ppc64le.tar.gz
+ENV PATH=/node-v13.5.0-linux-ppc64le/bin:$PATH
+
+RUN yum install -y  --enablerepo=epel libseccomp buildah \
```
Comment (johnmcollier) [inline src/pfe/Dockerfile_ppc64le]: The version of buildah bundled by default with CentOS is too backlevel for us to use, which is why we need to use ` ARG BUILDAH_RPM=https://cbs.centos.org/kojifiles/packages/buildah/1.11.4/4.el7/x86_64/buildah-1.11.4-4.el7.x86_64.rpm` on x86. 

We might need to work with the buildah team to an get up-to-date Buildah RPM for PPC.

Comment (vrushaliinamdar) [reply src/pfe/Dockerfile_ppc64le]: Yes, I tried using buildah 1.11.2 rpm for ppc64le, but does not work. It keeps on throwing missing dependency error for one or the other.

**Comment by vrushaliinamdar** [comment]
> I'm hesitant about including build scripts for appsody-controller in our repo. Shouldn't `build-appsody-controller.sh` be part of the https://github.com/appsody/controller repo?

We tried contacting them long back. But no response from them yet. Hence thought of building it from source for ppc64le.

**Comment by neeraj-laad** [comment]
Today, none of the Appsody components: CLI, controller, stacks are built or tested on ppc64le. So, I'm not sure what are we achieving by building individual components from source.

I think support for any new platform will need more thought and adjustments to processes so we can ensure ongoing support.

File: script/build.sh [hunk 2]
```diff
@@ -59,12 +59,17 @@ cp -r $DIR/docs ${SRC_DIR}/pfe/portal/
...
+#rm -f ${SRC_DIR}/pfe/extensions/codewind-appsody-extension-*.zip
 rm -f ${SRC_DIR}/pfe/extensions/codewind-odo-extension-*.zip
-curl -Lo ${SRC_DIR}/pfe/extensions/codewind-appsody-extension-9.9.9999.zip http://download.eclipse.org/codewind/codewind-appsody-extension/master/latest/codewind-appsody-extension-9.9.9999.zip
+#curl -Lo ${SRC_DIR}/pfe/extensions/codewind-appsody-extension-9.9.9999.zip http://download.eclipse.org/codewind/codewind-appsody-extension/master/latest/codewind-appsody-extension-9.9.9999.zip
 curl -Lo ${SRC_DIR}/pfe/extensions/codewind-odo-extension-9.9.9999.zip http://download.eclipse.org/codewind/codewind-odo-extension/master/latest/codewind-odo-extension-9.9.9999.zip
 
+# In case of ppc64le, download 'appsody' using ppc64le specific url and build 'appsody-controller' from source
+if [ "$ARCH" == "ppc64le" ]; then
+  $DIR/script/build-appsody-controller.sh
+fi
```
Comment (hhellyer) [inline script/build.sh]: I think based on the comment from @neeraj-laad https://github.com/eclipse/codewind/pull/1639#issuecomment-571989188 that support for Appsody in Codewind on ppc64le should probably be delivered when Appsody has full ppc64le support.
This PR will probably be much simpler if it just provides Codewind support for ppc64le without support for the Appsody extension.

**Comment by vrushaliinamdar** [comment]
> Today, none of the Appsody components: CLI, controller, stacks are built or tested on ppc64le. So, I'm not sure what are we achieving by building individual components from source.
> 
> I think support for any new platform will need more thought and adjustments to processes so we can ensure ongoing support.

@neeraj-laad  Recently the team has released the ppc64le specific appsody binary at 
https://github.com/appsody/appsody/releases/download/0.5.3/appsody-0.5.3-linux-ppc64le.tar.gz

is it feasible to have something similar for appsody-controller? or are there any steps through which we can easily achieve this?

**Comment by james-wallis** [comment]
@vrushaliinamdar I've made a change to `start.sh` to only pull down the version of `cwctl` that is going to be used. I took the logic from this PR and added it to the `if` statement see here: https://github.com/eclipse/codewind/pull/1654

Line for if statement: https://github.com/eclipse/codewind/pull/1654/files#diff-d0b2ef35f4d7cb0284dd5cf566b62f51R41-R57

**Comment by vrushaliinamdar** [comment]
> @vrushaliinamdar I've made a change to `start.sh` to only pull down the version of `cwctl` that is going to be used. I took the logic from this PR and added it to the `if` statement see here: #1654
> 
> Line for if statement: https://github.com/eclipse/codewind/pull/1654/files#diff-d0b2ef35f4d7cb0284dd5cf566b62f51R41-R57

Sure. Thanks.

**Comment by vrushaliinamdar** [comment]
Since building appsody-controller binary from source is not accepted, closing this PR and will raise a new one with the extension disabled.

**Comment by vrushaliinamdar** [comment]
Since building appsody-controller binary from source is not accepted, closing this PR and will raise a new one with the extension disabled.

**Pull request closed without merging**